### ISSUE TITLE
Cannot read property 'getBoundingClientRect' of null

### ISSUE THREAD

**Issue by ergonomicus** [tscanlin/tocbot#75]
@tscanlin  I get "Cannot read property 'getBoundingClientRect' of null" in console when I click on dropdown in my 
Bootstrap navigation. I'm pulling js and css from CDN. I have my code set as this:
```
tocbot.init({
    // Where to render the table of contents.
    tocSelector: '.help-sidebar',
    // Where to grab the headings to build the table of contents.
    contentSelector: '.help-main',
    // Which headings to grab inside of the contentSelector element.
    headingSelector: 'h1, h2, h3, h4, h5, h6',
});
```
The issue is somehow caused by my href="#!" on dropdown, As soon as I remove exclamation mark from href attribute - everything works fine.

Is this expected behaviour since I have `tocSelector` and `contentSelector` defined. I thought that these two were enough to isolate tocbot from the rest of the page.

Thanks.

**Comment by tscanlin** [comment]
Thanks for reporting this. This issue is the result of the smooth scrolling trying to listen to what it thinks is a hash link meant to scroll down the page but then it is unable to find the element. I can add some better error handling though and ignore hash links with an exclamation point.

**Comment by tscanlin** [comment]
`javascript:void(0)` or the class `no-smooth-scroll` would also fix btw, but shouldn't be required.

**Comment by tscanlin** [comment]
Fixed in ‘4.1.1’
